guard.cpp

```diff
@@ -7,43 +7,89 @@
 using namespace std;
 using namespace openpower::guard;
 
-void guardList()
+void printHeader(bool display)
+{
+    if (display)
+    {
+        std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
+    }
+    else
+    {
+        std::cout << "No Records to display" << std::endl;
+    }
+}
+
+void guardList(bool displayResolved)
 {
     auto records = getAll();
     if (!records.size())
     {
-        std::cout << "No Records to display" << std::endl;
+        std::cout << "No "
+                  << (displayResolved == true ? "resolved" : "unresolved")
+                  << " records to display" << std::endl;
         return;
     }
-    std::cout << "ID       | ERROR    |  Type  | Path " << std::endl;
+
+    bool isHeaderPrinted = false;
     for (const auto& elem : records)
     {
-        if ((elem.errType != GARD_Reconfig) &&
-            (elem.recordId != GUARD_RESOLVED))
+        // Not to print guard records with errorlog type set as GARD_Reconfig
```

Comment:
Please mention "Why" rather than saying "What doing" - IMO your code itself saying "What doing" (Not to print guard records with errorlog type set as GARD_Reconfig ==  if (elem.errType == GARD_Reconfig)"